### PR TITLE
Update co.elastic to org.opensearch

### PR DESCRIPTION
**Pull request by imRishN**
### Description
This PR updates namespace co.elastic to org.opensearch and fixes checkstyle issues.
 
### Issues Resolved
#12 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by mitalawachat — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

File: java-client/build.gradle.kts [hunk 1]
```diff
@@ -30,8 +30,8 @@ plugins {
 }
 
 java {
-    targetCompatibility = JavaVersion.VERSION_1_8
-    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_10
```
Comment (reta) [inline java-client/build.gradle.kts:33]: Why 1.8 was dropped? The minimum should be JDK-8 for 1.x

**Review by reta — CHANGES_REQUESTED** [review]

File: java-client/build.gradle.kts [hunk 2]
```diff
@@ -115,12 +115,12 @@ publishing {
 }
 
 dependencies {
-    val elasticsearchVersion = "7.16.0"
+
+    val opensearchVersion = "1.2.3"
```
Comment (reta) [inline java-client/build.gradle.kts:119]: 1.2.4  please

File: java-client/build.gradle.kts [hunk 3]
```diff
@@ -115,12 +115,12 @@ publishing {
...
     val jacksonVersion = "2.12.0"
```
Comment (reta) [inline java-client/build.gradle.kts:120]: 2.12.5 please (on par with core)

File: docs/installation.asciidoc [hunk 4]
```diff
@@ -24,7 +24,7 @@ available at https://snapshots.elastic.co/maven/.
 ["source","groovy",subs="attributes+"]
 --------------------------------------------------
 dependencies {
-    implementation 'co.elastic.clients:elasticsearch-java:{version}'
+    implementation 'org.opensearch.clients:elasticsearch-java:{version}'
```
Comment (reta) [inline docs/installation.asciidoc:27]: `elasticsearch-java ` should be changed

File: docs/installation.asciidoc [hunk 5]
```diff
@@ -42,7 +42,7 @@ dependencies:
 
   <dependencies>
     <dependency>
-      <groupId>co.elastic.clients</groupId>
+      <groupId>org.opensearch.clients</groupId>
       <artifactId>elasticsearch-java</artifactId>
```
Comment (reta) [inline docs/installation.asciidoc:46]: `elasticsearch-java ` should be changed

**Review by reta — CHANGES_REQUESTED** [review]

**Comment by reta** [comment]
@imRishN thanks for this change, I see 99% of the files in this pull request start with

```
// THIS CODE IS GENERATED. MANUAL EDITS WILL BE LOST.
//----------------------------------------------------
```

I am wondering, did you regenerated these files or changed them manually? Thank you.

File: java-client/build.gradle.kts (hunk 1 above)
Comment (imRishN) [reply java-client/build.gradle.kts:33]: Reverted back to 1.8

File: java-client/build.gradle.kts (hunk 2 above)
Comment (imRishN) [reply java-client/build.gradle.kts:119]: Thanks, updated

File: java-client/build.gradle.kts (hunk 3 above)
Comment (imRishN) [reply java-client/build.gradle.kts:120]: Updated

File: docs/installation.asciidoc (hunk 4 above)
Comment (imRishN) [reply docs/installation.asciidoc:27]: Updated

File: docs/installation.asciidoc (hunk 5 above)
Comment (imRishN) [reply docs/installation.asciidoc:46]: Updated

**Comment by imRishN** [comment]
> @imRishN thanks for this change, I see 99% of the files in this pull request start with
> 
> ```
> // THIS CODE IS GENERATED. MANUAL EDITS WILL BE LOST.
> //----------------------------------------------------
> ```
> 
> I am wondering, did you regenerate these files or changed them manually? Thank you.
I changed them manually. The above snippet has come from elasticsearch-java repo which will be removed in subsequent PRs. Thanks.

**Pull request merged**